src/Microsoft.VisualStudio.ProjectSystem.Managed/Text/StringHash.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Diagnostics.Contracts;
+
+namespace Microsoft.VisualStudio.Text
+{
+    internal static class StableStringHash
+    {
+        public static int GetDeterministicHashCode(this string str)
```

Comment:
I don't think we would want to use this hash function outside of the FUTDC. Perhaps move the method to `BuildUpToDateCheck.ItemHashing.cs`.

I'd also not make it an extension method, as otherwise it'll appear in IntelliSense in a bunch of places we don't want it.

Reply:
done